src/lib.rs

```diff
@@ -493,7 +577,12 @@ impl<'a> Builder<'a> {
                             .map(|ta| ta.contains(&*attr.name.local)) ==
                             Some(true);
                     if !whitelisted {
-                        false
+                        // If the class attribute is not whitelisted,
+                        // but there is a whitelisted set of allowed_classes,
+                        // do not strip out the class attribute.
+                        // Banned classes will be filtered later.
+                        &*attr.name.local == "class" &&
+                          self.allowed_classes.get(&*name.local).is_some()
```

Comment:
Maybe `contains_key()` instead. Too bad that the lookup will be performed twice.